Scripts/RemediationScripts/Remediate-EnableSecurityScanningForContainerRegistry.ps1

```diff
@@ -0,0 +1,716 @@
+    if ([String]::IsNullOrWhiteSpace($FilePath))
+    {
+        Write-Host "Fetching all Container Registry(s) in Subscription: $($context.Subscription.SubscriptionId)" -ForegroundColor $([Constants]::MessageType.Info)
+
+        # Get all Container Registry(s) in a Subscription
+        $containerRegistryDetails =  Get-AzResource -ResourceType Microsoft.ContainerRegistry/registries -ErrorAction Stop
+
+        # Seperating required properties
+        $containerRegistryDetails = $containerRegistryDetails | Select-Object @{N='ResourceId';E={$_.ResourceId}},
+                                                                          @{N='ResourceGroupName';E={$_.ResourceGroupName}},
+                                                                          @{N='ResourceName';E={$_.Name}},
+                                                                          @{N='ObjectId';E={$ObjectId}}
+    }
+    else
+    {
+        if (-not (Test-Path -Path $FilePath))
+        {
+            Write-Host "ERROR: Input file - $($FilePath) not found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+            break
+        }
+
+        Write-Host "Fetching all Container Registry(s) from [$($FilePath)]..." 
+
+        $containerRegistryResources = Import-Csv -LiteralPath $FilePath
+        $validcontainerRegistryResources = $containerRegistryResources| Where-Object { ![String]::IsNullOrWhiteSpace($_.ResourceId) }
+        
+        $validcontainerRegistryResources| ForEach-Object {
+            $resourceId = $_.ResourceId
+
+            try
+            {
+                $containerRegistryResource =  Get-AzResource -ResourceType Microsoft.ContainerRegistry/registries -ResourceGroupName $_.ResourceGroupName -Name $_.ResourceName -ErrorAction SilentlyContinue
+                $containerRegistryDetails += $containerRegistryResource
+            }
+            catch
+            {
+                Write-Host "Error fetching Container Registry(s) resource: Resource ID - $($resourceId). Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+            }
+        }
+    }
+
+    $totalContainerRegistry = ($containerRegistryDetails| Measure-Object).Count
+
+    if ($totalContainerRegistry -eq 0)
+    {
+        Write-Host "No Container Registry(s) found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        break
+    }
+  
+    Write-Host "Found [$($totalContainerRegistry)] Container Registry(s)." -ForegroundColor $([Constants]::MessageType.Update)
+    
+                                                                          
+    Write-Host $([Constants]::SingleDashLine)
+    # List for storing role assignment details.
+    $roleAssignmentDetails = @()
+
+    # List for storing sub level scanning identity role assignment details.
+    $subLevelRoleAssignmentDetails = @()
+
+    Write-Host "Fetching central reader security scanner role assignments..."
+
```

Comment:
Fetching role assignments of central security scanning identity...

Reply:
done